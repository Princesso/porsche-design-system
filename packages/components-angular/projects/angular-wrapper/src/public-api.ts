export { componentsReady } from '@porsche-design-system/components-js';
export * from './lib/components/barrel';
export * from './lib/types';
export { PorscheDesignSystemModule } from './porsche-design-system.module';
export type { PorscheDesignSystemModuleConfig } from './porsche-design-system.module';
export { ToastManager } from './toast-manager';
