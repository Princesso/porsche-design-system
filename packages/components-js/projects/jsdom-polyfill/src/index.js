// polyfills for features used by our components but unsupported within Jest (jsdom)
require('construct-style-sheets-polyfill');
require('intersection-observer');
require('matchmedia-polyfill');
require('matchmedia-polyfill/matchMedia.addListener');
require('scroll-behavior-polyfill');
require('whatwg-fetch');
global.ResizeObserver = require('@juggle/resize-observer').ResizeObserver;

// skip initial stylesheet fetch
window.PDS_SKIP_FETCH = true;

// workaround for WebComponentManager not working in JSDOM
require('./lib/loader.cjs')
  .defineCustomElements()
  .then(() => {
    window.PDS_SKIP_FETCH = false;

    // jsdom polyfill build does not load anything from CDN and also does not use our web components manager
    // therefore, we have to setup the document.porscheDesignSystem ourselves here
    // 'ROLLUP_REPLACE_VERSION' is replaced during build
    document.porscheDesignSystem = {
      ['ROLLUP_REPLACE_VERSION']: {
        readyResolve: () => {},
        isReady: () => Promise.resolve(),
      },
    };
  });

// TODO: ensure this isn't used in real browser?
