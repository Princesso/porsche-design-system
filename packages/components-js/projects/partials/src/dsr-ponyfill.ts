import { hydrateShadowRoots } from '@webcomponents/template-shadowroot';

if (!HTMLTemplateElement.prototype.hasOwnProperty('shadowRoot')) {
  hydrateShadowRoots(document.body);
}
