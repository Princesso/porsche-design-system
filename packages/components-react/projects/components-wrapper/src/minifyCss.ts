// TODO: remove :hover and :focus styles
export const minifyCss = (css: string): string =>
  css.replace(/\s\s+|\.\\(?=:)|[\n\\]+| (?={)|;(?=\s+})|(:|media)\s(?=.*;?)/g, '$1');
