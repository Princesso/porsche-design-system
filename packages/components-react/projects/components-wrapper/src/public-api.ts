// TODO: verify tree shaking
// @ts-ignore
if (!document.browser) {
  // @ts-ignore
  global.MutationObserver = class MutationObserverMock {
    // @ts-ignore
    constructor(callback: any) {}
    disconnect() {}
    // @ts-ignore
    observe(element, initObject) {}
  };
}

export { componentsReady } from '@porsche-design-system/components-js';
export * from './lib/components';
export * from './lib/types';
export { PorscheDesignSystemProvider } from './provider';
export { skipCheckForPorscheDesignSystemProviderDuringTests, useToastManager } from './hooks';
export { skipPorscheDesignSystemCDNRequestsDuringTests } from './utils';
