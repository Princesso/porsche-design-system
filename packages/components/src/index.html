<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Porsche Design System - Playground</title>
    <base href="/" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="icon" type="image/x-icon" href="favicon.ico" />
    <script nomodule src="build/porsche-design-system.js"></script>
    <script type="module" src="build/porsche-design-system.esm.js"></script>

    <style>
      /** {*/
      /*  margin: 0;*/
      /*  padding: 0;*/
      /*  box-sizing: border-box;*/
      /*}*/
    </style>
  </head>
  <body>
    <header>
      <h1>Porsche Design System - Playground</h1>
    </header>

    <main>
      <menu-toggle>
        <template shadowroot="open">
          <style>
            button {
              background: red;
            }
          </style>
          <button>
            <slot></slot>
          </button>
        </template>
        Open Menu
      </menu-toggle>

      <script>
        class MenuToggle extends HTMLElement {
          constructor() {
            super();

            // Detect whether we have SSR content already:
            if (this.shadowRoot) {
              console.log('DSR');
              // A Declarative Shadow Root exists!
              // wire up event listeners, references, etc.:
              const button = this.shadowRoot.firstElementChild;
              // button.addEventListener('click', toggle);
            } else {
              console.log('new SR');
              // A Declarative Shadow Root doesn't exist.
              // Create a new shadow root and populate it:
              // const shadow = this.attachShadow({ mode: 'open' });
              // shadow.innerHTML = `<button><slot></slot></button>`;
              // shadow.firstChild.addEventListener('click', toggle);
            }
          }
        }

        customElements.define('menu-toggle', MenuToggle);
      </script>
    </main>

    <script src="/dsr-ponyill.js"></script>

    <script>
      <!-- START: keep this snippet to log component lifecycle events -->
      ['componentWillLoad', 'componentDidLoad', 'componentWillUpdate', 'componentDidUpdate'].forEach((x) =>
        window.addEventListener(`stencil_${x}`, (e) => {
          const eventName = e.type + (e.type.includes('Did') ? ' ' : '');
          console.log(eventName, e.composedPath()[0].tagName.toLowerCase(), new Date().toISOString());
        })
      );
    </script>
    <!-- /END -->
  </body>
</html>
