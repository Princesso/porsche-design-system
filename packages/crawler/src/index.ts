import * as puppeteer from 'puppeteer';
import * as puppeteerConfig from '@porsche-design-system/shared/testing/jest-puppeteer.config';
import { crawlWebsites } from './crawl-websites';
import { removeOutdatedReports } from './helpers/fs-helper';
import { crawlerConfig } from '../constants';

const startBrowser = async (): Promise<void> => {
  const customerWebsites: string[] = [
    'https://www.porsche.com/germany',
    'https://finder.porsche.com/de/de-DE',
    'https://login.porsche.com/login/de/de_DE',
    'https://shop.porsche.com/de/de-DE',
    'https://www.porsche.com/swiss/de',
  ];

  const browser = await puppeteer.launch({
    ...puppeteerConfig,
    ...{
      // since some websites have different components depending on the window size & viewport,
      // we have to set the window size and viewport here in addition to shared config
      args: [`--window-size=${crawlerConfig.viewport.width},${crawlerConfig.viewport.height}`],
      defaultViewport: {
        width: crawlerConfig.viewport.width,
        height: crawlerConfig.viewport.height,
      },
    },
  });
  console.log('Removing outdated reports..');
  // removing old reports
  removeOutdatedReports(customerWebsites);
  console.log('Crawling websites..');
  // crawling all websites
  await crawlWebsites(browser, customerWebsites);
  console.log('Success - please check reports');
  await browser.close();
};

startBrowser().catch((err) => {
  throw new Error(`An error occurred during crawling: ${err}`);
});
