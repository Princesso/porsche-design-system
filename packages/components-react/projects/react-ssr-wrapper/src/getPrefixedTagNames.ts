import * as fromComponents from './lib/components';
import { camelCase } from 'change-case';
import { TagNameCamelCase } from '@porsche-design-system/shared';

export const getPrefixedTagNames = (): Record<TagNameCamelCase, any> => {
  return Object.entries(fromComponents).reduce(
    (res, [key, val]) => ({
      ...res,
      [camelCase(key)]: val,
    }),
    {} as Record<TagNameCamelCase, any>
  );
};
