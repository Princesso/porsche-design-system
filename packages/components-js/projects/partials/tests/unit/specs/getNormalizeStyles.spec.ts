import { getInitialStyles, getNormalizeStyles } from '../../../src';
import { render } from '@testing-library/react';

jest.mock('../../../src/shared');

describe('format: html', () => {
  it('should return normalize styles', () => {
    const result = getNormalizeStyles();
    // TODO: check if there is a better solution then snapshot?
    expect(result).toMatchSnapshot();
  });
});

describe('format: jsx', () => {
  it('should return normalize styles', () => {
    const { container } = render(getNormalizeStyles({ format: 'jsx' }));

    // TODO: check if there is a better solution then snapshot?
    expect(container.innerHTML).toMatchSnapshot();
  });
});
