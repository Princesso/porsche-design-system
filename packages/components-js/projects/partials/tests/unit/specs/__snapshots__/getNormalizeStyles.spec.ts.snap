// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format: html should return normalize styles 1`] = `"<style data-pds-normalize-styles>html{font-family:'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;line-height:1.15;text-size-adjust:none;-webkit-text-size-adjust:none}body{margin:0}article, aside, footer, header, nav, section, figcaption, figure, main{display:block}h1{font-size:2em;margin:0.67em 0}hr{box-sizing:content-box;height:0;overflow:visible}pre{font-family:monospace, monospace;font-size:1em}a{background:transparent;text-decoration-skip:objects;color:inherit;text-decoration:underline}a:active, a:hover{outline-width:0}abbr[title]{border-bottom:none;text-decoration:underline}b, strong{font-weight:700}code, kbp, samp{font-family:monospace, monospace;font-size:1em}dfn{font-style:italic}mark{background-color:#ff0;color:#000}small{font-size:80%}sub, sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sup{top:-0.5em}sub{bottom:-0.25em}audio, video{display:inline-block}audio:not([controls]){display:none;height:0}img{border-style:none;vertical-align:middle}svg:not(:root){overflow:hidden}button, input, optgroup, select, textarea{font-family:'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;font-size:1rem;line-height:calc(6px + 2.125ex);margin:0;text-size-adjust:none;-webkit-text-size-adjust:none;font-weight:400;font-style:normal;font-variant:normal;overflow-wrap:break-word;hyphens:auto}button, input{overflow:visible}button, select{text-transform:none}button, [type=reset], [type=submit]{-webkit-appearance:button}button::-moz-focus-inner, [type=button]::-moz-focus-inner, [type=reset]::-moz-focus-inner, [type=submit]::-moz-focus-inner{border-style:none;padding:0}button:-moz-focusring, [type=button]:-moz-focusring, [type=reset]:-moz-focusring, [type=submit]:-moz-focusring{outline:1px dotted ButtonText}fieldset{border:1px solid #c0c0c0;margin:0 2px;padding:0.35em 0.625em 0.75em}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal}progress{display:inline-block;vertical-align:baseline}textarea{overflow:auto}[type=checkbox], [type=radio]{box-sizing:border-box;padding:0}[type=number]::-webkit-inner-spin-button, [type=number]::-webkit-outer-spin-button{height:auto}[type=search]{appearance:textfield;outline-offset:-2px}[type=search]::-webkit-search-cancel-button, [type=search]::-webkit-search-decoration{appearance:none}::-webkit-file-upload-button{appearance:button;font:inherit}details, menu{display:block}summary{display:list-item}canvas{display:inline-block}template{display:none}[hidden]{display:none}em, i{font-style:normal}a, button, input, select, textarea{outline:1px solid transparent;outline-offset:1px}a::-moz-focus-inner{border:0}a:focus{outline-color:currentColor}a:focus:not(:focus-visible){outline-color:transparent}button::-moz-focus-inner{border:0}button:focus{outline-color:currentColor}button:focus:not(:focus-visible){outline-color:transparent}input::-moz-focus-inner{border:0}input:focus{outline-color:currentColor}input:focus:not(:focus-visible){outline-color:transparent}select::-moz-focus-inner{border:0}select:focus{outline-color:currentColor}select:focus:not(:focus-visible){outline-color:transparent}textarea::-moz-focus-inner{border:0}textarea:focus{outline-color:currentColor}textarea:focus:not(:focus-visible){outline-color:transparent}@media(hover:hover){a{transition:color var(--p-transition-duration, .24s) ease}a:hover{color:#ff0223}}</style>"`;

exports[`format: jsx should return normalize styles 1`] = `"<style data-pds-normalize-styles="">html{font-family:'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;line-height:1.15;text-size-adjust:none;-webkit-text-size-adjust:none}body{margin:0}article, aside, footer, header, nav, section, figcaption, figure, main{display:block}h1{font-size:2em;margin:0.67em 0}hr{box-sizing:content-box;height:0;overflow:visible}pre{font-family:monospace, monospace;font-size:1em}a{background:transparent;text-decoration-skip:objects;color:inherit;text-decoration:underline}a:active, a:hover{outline-width:0}abbr[title]{border-bottom:none;text-decoration:underline}b, strong{font-weight:700}code, kbp, samp{font-family:monospace, monospace;font-size:1em}dfn{font-style:italic}mark{background-color:#ff0;color:#000}small{font-size:80%}sub, sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sup{top:-0.5em}sub{bottom:-0.25em}audio, video{display:inline-block}audio:not([controls]){display:none;height:0}img{border-style:none;vertical-align:middle}svg:not(:root){overflow:hidden}button, input, optgroup, select, textarea{font-family:'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;font-size:1rem;line-height:calc(6px + 2.125ex);margin:0;text-size-adjust:none;-webkit-text-size-adjust:none;font-weight:400;font-style:normal;font-variant:normal;overflow-wrap:break-word;hyphens:auto}button, input{overflow:visible}button, select{text-transform:none}button, [type=reset], [type=submit]{-webkit-appearance:button}button::-moz-focus-inner, [type=button]::-moz-focus-inner, [type=reset]::-moz-focus-inner, [type=submit]::-moz-focus-inner{border-style:none;padding:0}button:-moz-focusring, [type=button]:-moz-focusring, [type=reset]:-moz-focusring, [type=submit]:-moz-focusring{outline:1px dotted ButtonText}fieldset{border:1px solid #c0c0c0;margin:0 2px;padding:0.35em 0.625em 0.75em}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal}progress{display:inline-block;vertical-align:baseline}textarea{overflow:auto}[type=checkbox], [type=radio]{box-sizing:border-box;padding:0}[type=number]::-webkit-inner-spin-button, [type=number]::-webkit-outer-spin-button{height:auto}[type=search]{appearance:textfield;outline-offset:-2px}[type=search]::-webkit-search-cancel-button, [type=search]::-webkit-search-decoration{appearance:none}::-webkit-file-upload-button{appearance:button;font:inherit}details, menu{display:block}summary{display:list-item}canvas{display:inline-block}template{display:none}[hidden]{display:none}em, i{font-style:normal}a, button, input, select, textarea{outline:1px solid transparent;outline-offset:1px}a::-moz-focus-inner{border:0}a:focus{outline-color:currentColor}a:focus:not(:focus-visible){outline-color:transparent}button::-moz-focus-inner{border:0}button:focus{outline-color:currentColor}button:focus:not(:focus-visible){outline-color:transparent}input::-moz-focus-inner{border:0}input:focus{outline-color:currentColor}input:focus:not(:focus-visible){outline-color:transparent}select::-moz-focus-inner{border:0}select:focus{outline-color:currentColor}select:focus:not(:focus-visible){outline-color:transparent}textarea::-moz-focus-inner{border:0}textarea:focus{outline-color:currentColor}textarea:focus:not(:focus-visible){outline-color:transparent}@media(hover:hover){a{transition:color var(--p-transition-duration, .24s) ease}a:hover{color:#ff0223}}</style>"`;
